<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:pfx10="http://www.tibco.com/CommonLE2/Private/Administration" xmlns:pfx="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns1="http://www.tibco.com/pe/DeployedVarsType" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:pfx5="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Severity" xmlns:tib="http://www.tibco.com/bw/xslt/custom-functions" xmlns:pfx4="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Category" xmlns:pfx3="http://www.tibco.com/CommonLE/namespace/Custom" xmlns:pfx2="http://xmlns.example.com/unique/default/namespace/1098735544074" xmlns:pfx9="http://schemas.tibco.com/bw/plugins/jdbc/5.0/jdbcExceptions" xmlns:pfx8="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:pfx7="http://www.tibco.com/CommonLE2/Public/Common" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:pfx6="http://www.tibco.com/CommonLE2/Public/Common/Dummy">
    <xsd:import namespace="http://www.tibco.com/CommonLE/namespace/Custom" schemaLocation="/Shared Resources/Schema Definitions/Custom/ExceptionCategory.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Category" schemaLocation="/SharedResources/SchemaDefinitions/Public/Pluggable/GUI/Administration/Configuration/ExceptionCategory.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Severity" schemaLocation="/SharedResources/SchemaDefinitions/Public/Pluggable/GUI/Administration/Configuration/ExceptionSeverity.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/Public/Common" schemaLocation="/SharedResources/SchemaDefinitions/Public/Common/ErrorMessage.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/private/Common" schemaLocation="/SharedResources/SchemaDefinitions/Private/Common/LocalExceptions.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/Private/Administration" schemaLocation="/SharedResources/SchemaDefinitions/Private/Administration/QueryUpdate.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/Public/Common/Dummy" schemaLocation="/SharedResources/SchemaDefinitions/Public/Pluggable/GUI/Common/DummyReturnMessage.xsd"/>
    <pd:name>Administration/DataAccess/Severity/MSSQL/CreateSeverity.process</pd:name>
    <pd:description>Creates a new exception severity</pd:description>
    <pd:startName>Start</pd:startName>
    <pd:startType ref="pfx10:SeverityUpdate"/>
    <pd:startX>30</pd:startX>
    <pd:startY>82</pd:startY>
    <pd:returnBindings>
        <pfx6:Dummy>
            <pfx6:StatusMessage>
                <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/ReservedIDs/DummyID"/>
            </pfx6:StatusMessage>
        </pfx6:Dummy>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endX>401</pd:endX>
    <pd:endY>86</pd:endY>
    <pd:errorSchemas>
        <Schema1 ref="pfx8:ApplicationIDNotFound"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1311738340027</pd:targetNamespace>
    <pd:activity name="Insert Exception Severity">
        <pd:type>com.tibco.plugin.jdbc.JDBCUpdateActivity</pd:type>
        <pd:resourceType>ae.activities.JDBCUpdateActivity</pd:resourceType>
        <pd:x>195</pd:x>
        <pd:y>91</pd:y>
        <config>
            <timeout>%%Database/MIG_Timeout%%</timeout>
            <commit>false</commit>
            <emptyStrAsNil>false</emptyStrAsNil>
            <jdbcSharedConfig>/SharedResources/Connections/Database/DBServerMSSQL.sharedjdbc</jdbcSharedConfig>
            <statement>Insert into exceptionseverity (ApplicationID, name, Description) 
Values (?, ?, ?)</statement>
            <Prepared_Param_DataType>
                <parameter>
                    <parameterName>ApplicationID</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
                <parameter>
                    <parameterName>SeverityName</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
                <parameter>
                    <parameterName>Description</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
            </Prepared_Param_DataType>
        </config>
        <pd:inputBindings>
            <jdbcUpdateActivityInput>
                <ApplicationID>
                    <xsl:value-of select="$Start/pfx10:SeverityUpdate/pfx10:ID"/>
                </ApplicationID>
                <SeverityName>
                    <xsl:value-of select="$Start/pfx10:SeverityUpdate/pfx10:Name"/>
                </SeverityName>
                <Description>
                    <xsl:value-of select="$Start/pfx10:SeverityUpdate/pfx10:Description"/>
                </Description>
            </jdbcUpdateActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="SQL Exception">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>86</pd:x>
        <pd:y>199</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=JDBCSQLException namespace=http://schemas.tibco.com/bw/plugins/jdbc/5.0/jdbcExceptions</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Generate Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>297</pd:x>
        <pd:y>200</pd:y>
        <config>
            <faultName>Schema1</faultName>
        </config>
        <pd:inputBindings>
            <pfx:ActivityInput>
                <message>
                    <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Common/Messages/ApplicationIDNotFound"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Common/Codes/ApplicationIDNotFound"/>
                </messageCode>
                <data>
                    <pfx8:ApplicationIDNotFound>
                        <pfx8:Message>
                            <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Common/Messages/ApplicationIDNotFound"/>
                        </pfx8:Message>
                        <pfx8:Code>
                            <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Common/Codes/ApplicationIDNotFound"/>
                        </pfx8:Code>
                    </pfx8:ApplicationIDNotFound>
                </data>
            </pfx:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Rethrow">
        <pd:type>com.tibco.pe.core.RethrowActivity</pd:type>
        <pd:resourceType>ae.activities.rethrow</pd:resourceType>
        <pd:x>86</pd:x>
        <pd:y>314</pd:y>
        <config/>
        <pd:inputBindings/>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Insert Exception Severity</pd:to>
        <pd:lineType>Straight</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Insert Exception Severity</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Straight</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>SQL Exception</pd:from>
        <pd:to>Generate Error</pd:to>
        <pd:xpathDescription>AppID Does Not Exist</pd:xpathDescription>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>xpath</pd:conditionType>
        <pd:xpath>$SQL-Exception/pfx9:JDBCSQLException/msgCode = $_globalVariables/ns1:GlobalVariables/Exceptions/DB/Codes/SQLIntegrityError and 
$SQL-Exception/pfx9:JDBCSQLException/sqlState = $_globalVariables/ns1:GlobalVariables/Exceptions/DB/Codes/SQLIntegrityState</pd:xpath>
    </pd:transition>
    <pd:transition>
        <pd:from>SQL Exception</pd:from>
        <pd:to>Rethrow</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>otherwise</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>